File: src/wsTripXML/Code/OTAClasses/OTACruisePackageAvail.vb
Option Strict Off
Option Explicit On 

Imports System.Xml.Serialization

Namespace wsTravelTalk.wmCruisePackageAvail

    '<remarks/>
    Public Enum PackageOptionPackageType

        '<remarks/>
        Pre

        '<remarks/>
        Post
    End Enum

    '<remarks/>
    Public Enum TransferOptionPackageType

        '<remarks/>
        Pre

        '<remarks/>
        Post

        '<remarks/>
        Round
    End Enum

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class SelectedCategory

        '<remarks/>
        <System.Xml.Serialization.XmlArrayItemAttribute(IsNullable:=False)> _
        Public CabinFilters() As CabinFilter

        '<remarks/>
        <System.Xml.Serialization.XmlElementAttribute("SelectedCabin")> _
        Public SelectedCabin() As SelectedCabin

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public BerthedCategoryCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public PricedCategoryCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public DeckName As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class SelectedCabin

        '<remarks/>
        <System.Xml.Serialization.XmlArrayItemAttribute(IsNullable:=False)> _
        Public CabinFilters() As CabinFilter

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Status As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public ReleaseDateTime As Date

        '<remarks/>
        <System.Xml.Serialization.XmlIgnoreAttribute()> _
        Public ReleaseDateTimeSpecified As Boolean

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CabinNumber As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public BedConfigurationCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public MaxOccupancy As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public DeclineIndicator As Boolean

        '<remarks/>
        <System.Xml.Serialization.XmlIgnoreAttribute()> _
        Public DeclineIndicatorSpecified As Boolean

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public HeldIndicator As Boolean

        '<remarks/>
        <System.Xml.Serialization.XmlIgnoreAttribute()> _
        Public HeldIndicatorSpecified As Boolean
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class CabinFilters

        '<remarks/>
        <System.Xml.Serialization.XmlElementAttribute("CabinFilter")> _
        Public CabinFilter() As CabinFilter
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class CabinFilter

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CabinFilterCode As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class SelectedSailing

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public VoyageID As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Start As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Duration As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public [End] As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public VendorCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public VendorName As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public ShipCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public ShipName As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute(DataType:="NMTOKEN")> _
        Public Status As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class SelectedFare

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public FareCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public GroupCode As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class InclusivePackageOption

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CruisePackageCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public InclusiveIndicator As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public StartDate As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class Currency

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CurrencyCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public DecimalPlaces As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class Budget

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public PricingType As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public MinPrice As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public MaxPrice As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public GuidelinePrice As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CurrencyCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public DecimalPlaces As String
    End Class

End Namespace

File: src/wsTripXML/Code/OTAClasses/OTACruisePackageAvail.cs
using System;
using System.Xml.Serialization;

namespace wsTripXML.wsTravelTalk.wmCruisePackageAvail
{

    // <remarks/>
    public enum PackageOptionPackageType
    {

        // <remarks/>
        Pre,

        // <remarks/>
        Post
    }

    // <remarks/>
    public enum TransferOptionPackageType
    {

        // <remarks/>
        Pre,

        // <remarks/>
        Post,

        // <remarks/>
        Round
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class SelectedCategory
    {

        // <remarks/>
        [XmlArrayItem(IsNullable = false)]
        public CabinFilter[] CabinFilters;

        // <remarks/>
        [XmlElement("SelectedCabin")]
        public SelectedCabin[] SelectedCabin;

        // <remarks/>
        [XmlAttribute()]
        public string BerthedCategoryCode;

        // <remarks/>
        [XmlAttribute()]
        public string PricedCategoryCode;

        // <remarks/>
        [XmlAttribute()]
        public string DeckName;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class SelectedCabin
    {

        // <remarks/>
        [XmlArrayItem(IsNullable = false)]
        public CabinFilter[] CabinFilters;

        // <remarks/>
        [XmlAttribute()]
        public string Status;

        // <remarks/>
        [XmlAttribute()]
        public DateTime ReleaseDateTime;

        // <remarks/>
        [XmlIgnore()]
        public bool ReleaseDateTimeSpecified;

        // <remarks/>
        [XmlAttribute()]
        public string CabinNumber;

        // <remarks/>
        [XmlAttribute()]
        public string BedConfigurationCode;

        // <remarks/>
        [XmlAttribute()]
        public string MaxOccupancy;

        // <remarks/>
        [XmlAttribute()]
        public bool DeclineIndicator;

        // <remarks/>
        [XmlIgnore()]
        public bool DeclineIndicatorSpecified;

        // <remarks/>
        [XmlAttribute()]
        public bool HeldIndicator;

        // <remarks/>
        [XmlIgnore()]
        public bool HeldIndicatorSpecified;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class CabinFilters
    {

        // <remarks/>
        [XmlElement("CabinFilter")]
        public CabinFilter[] CabinFilter;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class CabinFilter
    {

        // <remarks/>
        [XmlAttribute()]
        public string CabinFilterCode;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class SelectedSailing
    {

        // <remarks/>
        [XmlAttribute()]
        public string VoyageID;

        // <remarks/>
        [XmlAttribute()]
        public string Start;

        // <remarks/>
        [XmlAttribute()]
        public string Duration;

        // <remarks/>
        [XmlAttribute()]
        public string End;

        // <remarks/>
        [XmlAttribute()]
        public string VendorCode;

        // <remarks/>
        [XmlAttribute()]
        public string VendorName;

        // <remarks/>
        [XmlAttribute()]
        public string ShipCode;

        // <remarks/>
        [XmlAttribute()]
        public string ShipName;

        // <remarks/>
        [XmlAttribute(DataType = "NMTOKEN")]
        public string Status;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class SelectedFare
    {

        // <remarks/>
        [XmlAttribute()]
        public string FareCode;

        // <remarks/>
        [XmlAttribute()]
        public string GroupCode;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class InclusivePackageOption
    {

        // <remarks/>
        [XmlAttribute()]
        public string CruisePackageCode;

        // <remarks/>
        [XmlAttribute()]
        public string InclusiveIndicator;

        // <remarks/>
        [XmlAttribute()]
        public string StartDate;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class Currency
    {

        // <remarks/>
        [XmlAttribute()]
        public string CurrencyCode;

        // <remarks/>
        [XmlAttribute()]
        public string DecimalPlaces;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class Budget
    {

        // <remarks/>
        [XmlAttribute()]
        public string PricingType;

        // <remarks/>
        [XmlAttribute()]
        public string MinPrice;

        // <remarks/>
        [XmlAttribute()]
        public string MaxPrice;

        // <remarks/>
        [XmlAttribute()]
        public string GuidelinePrice;

        // <remarks/>
        [XmlAttribute()]
        public string CurrencyCode;

        // <remarks/>
        [XmlAttribute()]
        public string DecimalPlaces;
    }

}